File: backend/services/core/src/test/java/app/mnema/core/config/CacheConfigTest.java
package app.mnema.core.config;

import org.junit.jupiter.api.Test;
import org.springframework.cache.Cache;
import org.springframework.data.redis.connection.RedisConnectionFactory;

import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

class CacheConfigTest {

    @Test
    void cacheManagerBuildsRedisManager() {
        CacheConfig config = new CacheConfig();

        assertThat(config.cacheManager(mock(RedisConnectionFactory.class))).isNotNull();
    }

    @Test
    void cacheErrorHandlerSwallowsFailuresForKnownAndUnknownCaches() {
        CacheConfig config = new CacheConfig();
        Cache cache = mock(Cache.class);
        when(cache.getName()).thenReturn("core-media-resolve");

        var handler = config.cacheErrorHandler();
        RuntimeException ex = new RuntimeException("redis down");
        handler.handleCacheGetError(ex, cache, "key");
        handler.handleCachePutError(ex, cache, "key", "value");
        handler.handleCacheEvictError(ex, cache, "key");
        handler.handleCacheClearError(ex, cache);

        handler.handleCacheGetError(ex, null, "key");
    }
}

File: backend/services/user/src/test/kotlin/app/mnema/user/config/CacheConfigTest.kt
package app.mnema.user.config

import org.junit.jupiter.api.Assertions.assertNotNull
import org.junit.jupiter.api.Test
import org.mockito.Mockito.mock
import org.mockito.Mockito.`when`
import org.springframework.cache.Cache
import org.springframework.data.redis.connection.RedisConnectionFactory

class CacheConfigTest {

    private val config = CacheConfig()

    @Test
    fun `cache manager builds redis cache manager`() {
        val manager = config.cacheManager(mock(RedisConnectionFactory::class.java))

        assertNotNull(manager)
    }

    @Test
    fun `cache error handler logs and swallows cache exceptions`() {
        val cache = mock(Cache::class.java)
        `when`(cache.name).thenReturn("user-avatar-resolve")
        val handler = config.cacheErrorHandler()
        val exception = RuntimeException("redis unavailable")

        handler.handleCacheGetError(exception, cache, "key")
        handler.handleCachePutError(exception, cache, "key", "value")
        handler.handleCacheEvictError(exception, cache, "key")
        handler.handleCacheClearError(exception, cache)
    }
}
